pkg/types/baremetal/validation/platform.go

```diff
@@ -116,36 +117,117 @@ func validateHosts(hosts []*baremetal.Host, fldPath *field.Path) field.ErrorList
 	return hostErrs
 }
 
-func validateOSImages(p *baremetal.Platform, fldPath *field.Path) field.ErrorList {
-	platformErrs := field.ErrorList{}
+type contextKey int
 
-	validate := validator.New()
+const (
+	key contextKey = iota
+)
+
+type validationContext struct {
+	config       *types.InstallConfig
+	uniqueValues map[string]map[interface{}]struct{}
+	customErrs   map[string]error
+}
```

Comment:
i think this is mis-use of the context.

Reply:
`config` and `uniqueValues` are required contextual information to apply correctly the rules. `customErrs` is a necessary workaround required to build as much as possible a generic messaging layer that will adapt to the current one and will allow to reuse the pre-existing rules (and it could be removed once the issue https://github.com/go-playground/validator/issues/506 will be fixed). The usage of context in this case will allow more decoupling between the definition and calling of validation rules, that in theory could be pretty far each other (instead of closures)